lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -719,29 +719,44 @@ LogicalResult ExpandDimsOp::canonicalize(ExpandDimsOp op,
                                                  splat.getOperand());
     return mlir::success();
   }
-  // expand_dims(broadcast) -> broadcast(expand_dims)
+  // expand_dims(broadcast(x)) -> broadcast(expand_dims(x))
   //
-  // On it's own this doesn't do much, but consider
+  // On its own this doesn't do much, but consider
   //    broadcast(expand_dims(broadcast))
   // -> broadcast(broadcast(expand_dims))
   // -> broadcast(expand_dims)
+  //
+  // Note we can't do this if x is a scalar, because expandDims requires a
+  // tensor input.  (We could change it to broadcast(expand_dims(splat(x))), but
+  // it's unclear that's better.)  broadcast currently accepts scalar inputs,
```

Comment:
> We could change it to broadcast(expand_dims(splat(x))), but it's unclear that's better.

You could just emit `splat(x)` here.

Reply:
Agreed, but I wanted to leave this until we got rid of this special case altogether (as you suggested).